src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -1335,27 +1345,22 @@ public synchronized void setSkin(@Nullable PlayerSkin skin) {
         DestroyEntitiesPacket destroyEntitiesPacket = new DestroyEntitiesPacket();
         destroyEntitiesPacket.entityIds = new int[]{getEntityId()};
 
-        final PlayerInfoPacket removePlayerPacket = getRemovePlayerToList();
-        final PlayerInfoPacket addPlayerPacket = getAddPlayerToList();
-
         RespawnPacket respawnPacket = new RespawnPacket();
         respawnPacket.dimensionType = getDimensionType();
         respawnPacket.gameMode = getGameMode();
         respawnPacket.isFlat = levelFlat;
 
-        playerConnection.sendPacket(removePlayerPacket);
+        for (TabList tabList : MinecraftServer.getTabListManager().getTabLists()) {
+            if (tabList.getDisplayedPlayers().contains(this)) {
+                tabList.removeDisplayedPlayer(this);
+                tabList.addDisplayedPlayer(this);
+            } else {
+                tabList.addDisplayedPlayer(this);
+                tabList.removeDisplayedPlayer(this);
+            }
+        }
```

Comment:
Is it possible that with how the networking side of things is set up that these packets could be sent out of order? @TheMode 

Also, if the player isn't in a given tab list, why do you bother adding and removing them? Shouldn't we still be sending the packet to all viewers here in case they aren't in a tab list together but can still view a given player?